client/api/disk/v1alpha1/api.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+
+package v1alpha1;
+
+service Disk {
+    // ListDiskLocations returns locations <Adapter, Bus, Target, LUN ID> of all
+    // disk devices enumerated by the host
+    rpc ListDiskLocations(ListDiskLocationsRequest) returns (ListDiskLocationsResponse) {}
+
+    // PartitionDisk initializes and partitions a disk device (if the disk has not
+    // been partitioned already) and returns the resulting volume device ID
+    rpc PartitionDisk(PartitionDiskRequest) returns (PartitionDiskResponse) {}
+
+    // Rescan refreshes the host's storage cache
+    rpc Rescan(RescanRequest) returns (RescanResponse) {}
+}
```

Comment:
do we also need format and initilize_disk apis?

Reply:
As discussed during the meeting today, `format` is taken care of as part of the volume APIs - https://github.com/kubernetes-csi/csi-proxy/blob/master/client/api/volume/v1alpha1/api.proto#L16 and disk initialization is part of Disk Partitioning: https://github.com/kubernetes-csi/csi-proxy/pull/15/files#diff-87dc7f1dc074717f7df0d9698f32ce1fR51